datastream-jdbc-connector/src/main/java/com/linkedin/datastream/connectors/jdbc/translator/JdbcCommon.java

```diff
@@ -335,9 +331,14 @@ public static long convertToAvroStream(final ResultSet rs, final OutputStream ou
                             // The cause of the exception is not known, but we'll fall back to call getObject() and handle any "real" exception there
                             value = rs.getObject(i);
                         }
-                    } else if (javaSqlType == microsoft.sql.Types.DATETIMEOFFSET) { // -155 represents TSQL Datetimeoffset type
+                    } else if (javaSqlType == TIMESTAMP // 93
+                            || javaSqlType == TIMESTAMP_WITH_TIMEZONE // 2014
+                            || javaSqlType == microsoft.sql.Types.SMALLDATETIME // -150 represents TSQL smalldatetime type
+                            || javaSqlType == microsoft.sql.Types.DATETIME // -151 represents TSQL datetime type
+                            || javaSqlType == microsoft.sql.Types.DATETIMEOFFSET) { // -155 represents TSQL Datetimeoffset type
+
                         try {
-                            // Convert datetimeoffset to UTC then convert to timestamp millis
+                            // Convert datetime, datetime2 and datetimeoffset to UTC then convert to timestamp millis
```

Comment:
This is a good approach. But you can't really convert all types except DATETIMEOFFSET to UTC as you don't know what is the source timezone is.

You don't need to change anything here. But you will have to add an additional property in the schema that allows the consumer identify what is the source field type so that they can properly infer the timezone based on the set property.

See my comment on schema section.